File: lab11_12/WebDriverAutomation/Pages/LoginPage.cs
using OpenQA.Selenium;
using SeleniumExtras.PageObjects;

namespace WebDriverAutomation.Pages;

public class LoginPage
{
    private const string BASE_URL = "https://www.comicskrama.by/client_account/contacts/new";

    [FindsBy(How = How.XPath, Using = "//input[@id='client_name']")]
    private IWebElement _fullNameInput;

    [FindsBy(How = How.XPath, Using = "//input[@id='client_phone']")]
    private IWebElement _phoneNumberInput;

    [FindsBy(How = How.XPath, Using = "//input[@id='client_email']")]
    private IWebElement _emailInput;

    [FindsBy(How = How.XPath, Using = "//input[@id='client_password']")]
    private IWebElement _passwordInput;

    [FindsBy(How = How.XPath, Using = "//*[@id='client_password_confirmation']")]
    private IWebElement _repeatPasswordInput;

    [FindsBy(How = How.XPath, Using = "//input[@id='new_client']/div[8]/button")]
    private IWebElement _signUpButton;

    private IWebDriver _driver;

    public LoginPage(IWebDriver driver)
    {
        this._driver = driver;
        PageFactory.InitElements(this._driver, this);
    }
            
    public void OpenPage()
    {
        _driver.Navigate().GoToUrl(BASE_URL);
        Console.WriteLine("Login page opened");
    }

    public void SignUp(string password, string email, string phoneNumber, string fullName)
    {
        _fullNameInput.SendKeys(fullName);
        _phoneNumberInput.SendKeys(phoneNumber);
        _emailInput.SendKeys(email);
        _passwordInput.SendKeys(password);
        _repeatPasswordInput.SendKeys(password);
        _signUpButton.Click();
    }
}

File: lab_final/src/test/java/page/LoginPage.java
package page;

import model.*;
import org.apache.logging.log4j.*;
import org.openqa.selenium.*;
import org.openqa.selenium.support.*;


public class LoginPage extends AbstractPage{
    private final String BASE_URL = "https://www.comicskrama.by/client_account/contacts/new";
    private final Logger logger = LogManager.getRootLogger();

    @FindBy(xpath = "//input[@id='client_name']")
    private WebElement fullNameInput;

    @FindBy(xpath = "//input[@id='client_phone']")
    private WebElement phoneNumberInput;

    @FindBy(xpath = "//input[@id='client_email']")
    private WebElement emailInput;

    @FindBy(xpath = "//input[@id='client_password']")
    private WebElement passwordInput;

    @FindBy(xpath = "//*[@id='client_password_confirmation']")
    private WebElement repeatPasswordInput;

    @FindBy(xpath = "//input[@id='new_client']/div[8]/button")
    private WebElement signUpButton;

    public LoginPage(WebDriver driver){
        super(driver);
        PageFactory.initElements(this.driver, this);
    }

    public LoginPage openPage(){
        driver.navigate().to(BASE_URL);
        return this;
    }

    public String singUp(User user){

        fullNameInput.sendKeys(user.getUsername());
        phoneNumberInput.sendKeys(user.getPhoneNumber());
        emailInput.sendKeys(user.getEmail());
        passwordInput.sendKeys(user.getPassword());
        repeatPasswordInput.sendKeys(user.getPassword());
        signUpButton.click();

        logger.info("Registration perfomed");

        return getUrlLoginPage();
    }

    private String getUrlLoginPage(){
        return driver.getCurrentUrl();
    }
}
